### PR TITLE
Add unsafeHas

### PR DESCRIPTION
**Pull request by natefaubion**

### CODE REVIEW

**Comment by paulyoung** [comment]
@natefaubion could you help me understand what makes this "unsafe"?

File: src/Data/Record/Unsafe.js [hunk 1]
```diff
@@ -24,3 +24,7 @@ exports.unsafeDeleteFn = function(label, rec) {
   }
   return copy;
 };
+
+exports.unsafeHasFn = function(label, rec) {
+  return {}.hasOwnProperty.call(rec, label);
```
Comment (paulyoung) [inline src/Data/Record/Unsafe.js:29]: How about using `Object.prototype.hasOwnProperty.call(rec, label)` so that a new object isn't created?

Comment (natefaubion) [reply src/Data/Record/Unsafe.js:29]: It's using the same code the compiler generates in record updates. Engines will optimize this out.

**Comment by natefaubion** [comment]
It's unsafe because it violates referential transparency. The type signature says it works `forall r. { | r }`, but you are able to observe more than that.

**Comment by natefaubion** [comment]
Maybe referential transparency isn’t the right word? There isn’t a use for this function in safe code since the whole point of records is that you know the fields statically, making this kind of check unnecessary. The only use case for it is if you are making a dynamic assertion about records, which I would consider inherently unsafe. Maybe “disingenuousHas” is a better name? :D

**Comment by paf31** [comment]
Thanks!

**Pull request merged**